include/subgroup/tile/impl/store_xe.hpp

```diff
@@ -544,6 +547,61 @@ tile_store(tile_t& tile, payload_t& payload) {
   }
 }
 
+/// @brief Is the func storing data from register file to unaligned global
+/// memory surface. store a rectangular region (X,Y)..(X+W,Y+H) into memory from
+/// registers.
+/// @tparam tile_t Is the tile_t struct contains registers
+/// These registers will be the source of store operation.
+/// @tparam payload_t Is the mem_payload_t struct describing the memory info
+/// payload indicates the destination of store operation.
+/// @tparam L1 Is the cache hint for L1 cache.
+/// @tparam L3 Is the cache hint for L3 cache.
+/// @param tile Is the tile object with type tile_t, contains the data to be
+/// stored.
+/// @param payload Is the payload object with type payload_t. Contains all the
+/// information for stores.
+/// @return No return, update in place.
+template <
+    cache_hint L1 = cache_hint::write_back,
+    cache_hint L2 = cache_hint::write_back,
+    typename tile_t,
+    typename payload_t>
+__XETLA_API typename std::enable_if_t<
+    detail::check_store_type<tile_t, payload_t>::is_global_block_2d &&
+    !detail::check_store_type<tile_t, payload_t>::is_lsc_scatter &&
+    payload_t::arch_tag <= gpu_arch::XeHpg>
```

Comment:
is this can be replaced by " ! arch_has_2d_load_store<payload_T::arch_tag>" ?

Reply:
fixed  https://github.com/intel/neural-speed/pull/256/commits/0090354dd214c8509b9393fad605ab84fb361280